### PR TITLE
feat: generate deployment artifacts with forc-deploy

### LINKED ISSUE #4744: forc-deploy should generate deployment artifacts
Deployments are really easy to lose with the current state of forc-deploy. We should generate some artifacts for each deployment, so that we can keep track of previous deployments. We should look into foundry and hardhat to see what info they are serializing for this purpose.

#4723 have some details about this issue as well.

### PR DESCRIPTION
**Pull request by kayagokalp**
## Description
closes #4744.

This PR adds deployment artifact serialization to forc-deploy. Basically each deployment info is saved as a json file in the out folder. So that users wont lose precious deployment information after closing their terminal.

An example deployment artifact can be seen below:
```json
{
  "transaction_id": "0xec27bb7a4c8a3b8af98070666cf4e6ea22ca4b9950a0862334a1830520012f5d",
  "salt": "0x9e35d1d5ef5724f29e649a3465033f5397d3ebb973c40a1d76bb35c253f0dec7",
  "network_endpoint": "http://127.0.0.1:4000",
  "chain_id": 0,
  "contract_id": "0x767eeaa7af2621e637f9785552620e175d4422b17d4cf0d76335c38808608a7b",
  "deployment_size": 68,
  "deployed_block_id": "0x915c6f372252be6bc54bd70df6362dae9bf750ba652bf5582d9b31c7023ca6cf"
}                                                                                                                                                                                                                     
```
Since serde serialization for `ContarctId` and other deployment related client/vm primitives do not emit 0x at the beginning i had to convert them to `String`s 0x prefixed. I feel like we can change human readable serialization to include 0x to all hex fields.


## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by kayagokalp** [comment]
After merging this and #4771, we will be closing #4723 and #4745

File: forc-plugins/forc-client/src/op/deploy.rs [hunk 1]
```diff
@@ -254,6 +293,28 @@ pub async fn deploy_pkg(
                 info!("\nNetwork: {node_url}");
                 info!("Contract ID: 0x{contract_id}");
                 info!("Deployed in block {}", &block_id);
+
+                // Create a deployment articact.
```
Comment (JoshuaBatty) [inline forc-plugins/forc-client/src/op/deploy.rs:297]:
```suggestion
                // Create a deployment artifact.
```

**Review by JoshuaBatty — CHANGES_REQUESTED** [review]

File: forc-plugins/forc-client/src/op/deploy.rs (hunk 1 above)
Comment (kayagokalp) [reply forc-plugins/forc-client/src/op/deploy.rs:297]: 🤦 Thanks!

**Review by sdankel — DISMISSED** [review]

**Comment by JoshuaBatty** [comment]
Nice work, this will be super handy for users. The code looks good. Would you mind also adding this to the docs, maybe showing the example of an generated artifacts and some information about where the json is saved etc..?

**Comment by kayagokalp** [comment]
Added docs with example artifact, should be good to go!

File: docs/book/src/forc/plugins/forc_client/index.md [hunk 2]
```diff
@@ -63,3 +63,19 @@ Also the default value of the "gas price" parameter is 0 for both `forc-deploy`
 ```sh
 forc-deploy --node-url https://beta-3.fuel.network/graphql --gas-price 1
 ```
+
+## Deployment Artifacts
+
+forc-deploy stores deployment details in 'out/deployments' folder in project root directory for each deployment. An example deployment artifact can be seen below:
```
Comment (JoshuaBatty) [inline docs/book/src/forc/plugins/forc_client/index.md:69]:
```suggestion
forc-deploy saves the details of each deployment in the `out/deployments` folder within the project's root directory. Below is an example of a deployment artifact:
```

**Review by eureka-cpu — APPROVED** [review]
LGTM

**Pull request merged**